api/src/main/java/bio/terra/tanagra/service/query/QueryService.java

```diff
@@ -127,6 +155,62 @@ Query createQuery(EntityDataset entityDataset) {
     return queryBuilder.build();
   }
 
+  @VisibleForTesting
+  Query createQuery(EntityCounts entityCounts) {
+    ImmutableList<Selection> groupByFields =
+        entityCounts.groupByAttributes().stream()
+            .map(
+                attribute ->
+                    Selection.SelectExpression.builder()
+                        .expression(
+                            AttributeExpression.create(
+                                AttributeVariable.create(
+                                    attribute, entityCounts.primaryEntity().variable())))
+                        // set the attribute alias to empty string, because we can't use the AS
+                        // keyword in a GROUP BY clause
+                        .name("")
+                        .build())
+            .collect(ImmutableList.toImmutableList());
+
+    // select fields for a count query include: COUNT(tableAlias), [group by attributes],
+    // [additional selected attributes]
+    List<Selection> selectFields = new ArrayList<>();
+    selectFields.add(
+        Selection.Count.builder()
+            .name(COUNT_ALIAS)
+            .entityVariable(entityCounts.primaryEntity())
+            .build());
+    entityCounts.groupByAttributes().stream()
```

Comment:
Looks like the mappers are identical here, so I would probably go for:

```
Stream.concat(
  entityCounts.groupByAttributes().stream(),
  entityCounts.additionalSelectedAttributes().stream()
).map(
  ...
).forEach(selectedFields::add);
```

Reply:
Ah that looks nicer, thanks!